test/test_cpp_extensions_mtia_backend.py

```diff
@@ -11,20 +11,18 @@
     IS_ARM64,
     IS_LINUX,
     skipIfTorchDynamo,
-    TEST_CUDA,
     TEST_PRIVATEUSE1,
-    TEST_XPU,
 )
-from torch.utils.cpp_extension import CUDA_HOME, ROCM_HOME
 
 
-# define TEST_ROCM before changing TEST_CUDA
-TEST_ROCM = TEST_CUDA and torch.version.hip is not None and ROCM_HOME is not None
-TEST_CUDA = TEST_CUDA and CUDA_HOME is not None
+# This TestCase should be mutually exclusive with other backends.
+HAS_CUDA = torch.backends.cuda.is_built()
+HAS_XPU = torch.xpu._is_compiled()
+HAS_MPS = torch.backends.mps.is_built()
```

Comment:
@albanD We can't use `torch.accelerator.current_accelerator` here to determine the build type, because calling it before the MTIA backend extension is loaded will cause `hasMTIA` to always return `False`.
See [DETECT_RUNTIME_ACCELERATOR(MTIA)](https://github.com/pytorch/pytorch/blob/05365e380d01683d3f415cd7997c7425a80d0427/aten/src/ATen/DeviceAccelerator.cpp#L27) for reference.